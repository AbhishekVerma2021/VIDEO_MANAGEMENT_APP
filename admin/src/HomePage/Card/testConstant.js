export const data = {
    "link": "https://www.youtube.com/watch?v=e64JTo7LMmY",
    "title": "This is a title",
    "description": "sadn asd sad  saf rewef wefew f wef we fw ef wef we f",
    "_id": "643d8663cdcd1ac35198e3ae",
    "date": "2023-04-17"
}