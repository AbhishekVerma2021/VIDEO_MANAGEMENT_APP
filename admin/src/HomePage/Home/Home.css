.cardHome {
    margin: 0vw 0vw 0.5vw 0vw;
    padding-top: 12vh;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: auto;
    column-gap: 5vh;
    row-gap: 10vh;
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: 100vw 100vh;
    position: relative;
}



.message {
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}

h2 {
    color: #5e6062;
    margin: 0;
}

@media only screen and (max-width: 400px) {
    .cardHome {
        display: grid;
        grid-template-columns: 1fr;
    }
}