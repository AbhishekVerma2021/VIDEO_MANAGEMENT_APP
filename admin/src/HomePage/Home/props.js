import { getData, deletePost, validateUserToken, searchResultData } from '../../Redux/actions';

const mapStateToProps = state => {
    // console.
    return {
        postCardData: state.postCardData,
        searchResultPostDetails: state.searchResultPostDetails,
        loggedInUserData: state.loggedInUserData
    };
};
const mapDispatchToProps = dispatch => {
    return {
        getData: (page, limit, history) => dispatch(getData(page, limit, history)),
        deletePost: (_id) => dispatch(deletePost(_id)),
        searchResultData: (searchString, history) => dispatch(searchResultData(searchString, history))
        // validateUserToken: (token) => dispatch(validateUserToken(token))
    };
};

export { mapStateToProps, mapDispatchToProps }