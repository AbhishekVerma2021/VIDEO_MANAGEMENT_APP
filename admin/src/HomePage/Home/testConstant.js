export const postCardData = [{
    "date": "2023-04-18T15:29:54.942Z",
    "description": "Unleash your inner power and unlock your full potential to achieve your dreams and goals.",
    "link": "https://www.youtube.com/watch?v=1jhY2fhAsh8",
    "title": "Believe in Yourself, Achieve Your Dreams",
    "_id": "643e66dc66f6cc577576d0aa"
}]

export const searchResultPostDetails = [{
    "date": "2023-04-18T15:29:54.942Z",
    "description": "Empower your mindset with positive thinking, mindset shifts, and personal development.",
    "link": "https://www.youtube.com/watch?v=X2X_qbyVU84",
    "title": "Empowering Your Mindset",
    "_id": "643e66dc66f6cc577576d0a9"
}] 