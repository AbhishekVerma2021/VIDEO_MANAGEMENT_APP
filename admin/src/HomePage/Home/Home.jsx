import React from 'react'
import NavBar from '../../NavBar/NavBar'
import Card from '../Card/Card'
import './Home.css'
import 'react-toastify/dist/ReactToastify.css';
// import { connect } from 'react-redux'
// import { getData, deletePost, validateUserToken } from '../../Redux/actions'
import { ToastContainer, toast } from 'react-toastify';
import axios from 'axios';
// import { } from '../../Redux/actions';
// import { withRouter } from 'react-router-dom';

class Home extends React.Component {

  constructor(props) {
    super(props)

  }
  componentDidMount() {
    axios.get('http://localhost:8000/validateToken', {
      headers: {
        authorization: `Bearer ${localStorage.getItem('token')}`
      }
    }).then(res => {
      console.log(res)
      this.props.getData(this.props.history)
    })
      .catch(error => {
        console.log(error)
        toast.error(error.response.data, {
          position: toast.POSITION.BOTTOM_RIGHT,
          autoClose: 3000,
        })
        this.props.history.push('/'); localStorage.clear()
      })
  }


  handleDelete = (_id) => {
    const token = localStorage.getItem('token');
    if (!token) {
      this.props.history.push('/')
    }
    const { deletePost } = this.props;
    deletePost(_id);
  }


  render() {
    const { postCardData, searchResultPostDetails } = this.props
    console.log("SEARCH_DATA", this.props.searchResultPostDetails)
    console.log("POST_CARD_DATA", this.props.postCardData)
    // console.log(postCardData)
    return <>
      <NavBar />
      {searchResultPostDetails.length !== 0 ?
        (<div>
          {
            searchResultPostDetails && searchResultPostDetails.length === 0 ? <div className='message'><h2>No Video Search availble. Please upload one.</h2></div> :
              <div className="cardHome">
                {
                  Object.values(searchResultPostDetails).map((details) => {
                    return (<Card data={details} onDelete={() => {
                      this.handleDelete(details._id)
                    }} />)
                  })
                }
              </div>
          }
          <ToastContainer />
        </div>) : (<div>
          {
            postCardData && postCardData.length === 0 ? <div className='message'><h2>No Video availble. Please upload one.</h2></div> :
              <div className="cardHome">
                {
                  Object.values(postCardData).map((details) => {
                    return (<Card data={details} onDelete={() => {
                      this.handleDelete(details._id)
                    }} />)
                  })
                }
              </div>
          }
          <ToastContainer />
        </div>)
      }
    </>
  }
}

export default Home;