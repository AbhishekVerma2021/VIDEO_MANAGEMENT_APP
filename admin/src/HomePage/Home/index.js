import { connect } from "react-redux";
import Home from "./Home";

import { mapStateToProps, mapDispatchToProps } from "./props";
import { withRouter } from 'react-router-dom';

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Home));