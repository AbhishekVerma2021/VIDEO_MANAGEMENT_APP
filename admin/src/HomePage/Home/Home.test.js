import React from 'react';
import Home from './Home';
import { shallow, configure } from 'enzyme';
import Adapter from "@cfaester/enzyme-adapter-react-18";
configure({ adapter: new Adapter() });
import { postCardData, searchResultPostDetails } from './testConstant';
import Home from './Home';

describe('Test <Home/>', () => {
    it('Should render Home componnet', () => {
        const component = shallow(< Home postCardData={postCardData} searchResultPostDetails={searchResultPostDetails} />);
        const tree = component.debug();
        expect(tree).toMatchSnapshot();
    })
})