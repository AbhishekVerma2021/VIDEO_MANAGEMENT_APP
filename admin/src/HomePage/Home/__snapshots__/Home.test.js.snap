// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Home/> Should render Home componnet 1`] = `
"<Fragment>
  <withRouter(Connect(NavBar)) />
  <div>
    <div className=\\"cardHome\\">
      <Card data={{...}} onDelete={[Function: onDelete]} />
    </div>
    <ToastContainer position=\\"top-right\\" transition={[Function (anonymous)]} autoClose={5000} closeButton={[Function: E]} pauseOnHover={true} pauseOnFocusLoss={true} closeOnClick={true} draggable={true} draggablePercent={80} draggableDirection=\\"x\\" role=\\"alert\\" theme=\\"light\\" />
  </div>
</Fragment>"
`;
